### PR TITLE
Added a UUID field

### PR DESCRIPTION
**Pull request by thedrow**
There was no UUIDField so I added it.
I hope the code here is good enough for merge and release.

### CODE REVIEW

File: odin/fields/__init__.py [hunk 1]
```diff
@@ -805,3 +806,33 @@ class IPv46Field(StringField):
...
+            return value
+        elif isinstance(value, bytes):
+            if len(value) == 16:
+                return uuid.UUID(bytes=value)
+            else:
+                value = value.decode('utf-8')
+        elif isinstance(value, int):
+            return uuid.UUID(int=value)
+        elif isinstance(value, (tuple, list)):
+                return uuid.UUID(fields=value)
```
Comment (timsavage) [inline odin/fields/__init__.py]: Is there an extra indent here?

File: odin/fields/__init__.py [hunk 2]
```diff
@@ -805,3 +806,33 @@ class IPv46Field(StringField):
...
+
+    Validates that the string represents a universally unique identifier.
+    """
+    data_type_name = "UUID"
+
+    def __init__(self, **options):
+        options.setdefault('validators', []).append(validate_uuid)
+        super(UUIDField, self).__init__(**options)
+
+    def to_python(self, value):
```
Comment (timsavage) [inline odin/fields/__init__.py]: This method needs to handle ValueError exceptions during decode, and type conversion.

File: odin/validators.py [hunk 3]
```diff
@@ -249,3 +250,47 @@ def validate_domain_part(self, domain_part):
                 pass
 
 validate_email_address = EmailValidator()
+
+
+class UUIDValidator(object):
```
Comment (timsavage) [inline odin/validators.py]: I'm not sure this validator is required it is essentially doing what fields do (eg decode the value). The field.to_python step occurs prior to validation so the only type that should be supplied to a validator is a UUID. The email/ip address fields are all strings that are required to be in a specific format.

File: AUTHORS [hunk 4]
```diff
@@ -1,5 +1,6 @@
-* Tim Savage <tim@savage.company>
-
-Contributors:
-
-* thedrow for a patch that adds tests that exposes a bug in handling of nested resources.
+Omer Katz <omer.drow@gmail.com>
```
Comment (timsavage) [inline AUTHORS]: Think you have accidentally included this file (is generated by PBR)

File: odin/validators.py (hunk 3 above)
Comment (thedrow) [reply odin/validators.py]: So all the validation should occur in `to_python()`?

Comment (timsavage) [reply odin/validators.py]: That's correct. Validators are called on values that have already been processed by `to_python`.

**Comment by thedrow** [comment]
@timsavage Right, this is ready. I'd appreciate a release to PyPi after merging this.

**Pull request merged**